### DISCUSSION TITLE
promptInstall callback

### DISCUSSION THREAD

**Discussion by drogon98** [ShafSpecs/remix-pwa#150] [Ideas]
I think it would be better if i someone can keep count of installations. The only way to achieve this is by attaching a callback to the promptInstall method. Or is there a way of doing this currently?

**Comment by ShafSpecs** [comment]
> I think it would be better if i someone can keep count of installations. The only way to achieve this is by attaching a callback to the promptInstall method. All is there a way of doing this currently?

When you say keep count, do you mean server side or locally? Could you please provide a psuedo code of what you mean?

**Reply by drogon98** [reply]
Think of google playstore downloads count. 

```
const callback = async ()=>{
// Make a request to update db with 1 installation
}
```

// promptInstall can have a callback argument
`promptInstall(cbk?:()=>Promise<void>)`

// In the promptInstall i can call the callback based on user choice
```
const promptInstall = (cbk?:()=>Promise<void>) => {
    if (installPromptEvent) {
      const installEvent = (installPromptEvent as any)
      const result = await installEvent.prompt();
      if(result==="accepted"){
         await cbk?.()
     } 
      setInstallPromptEvent(null);
    }
  };
```

**Reply by ShafSpecs** [reply]
👍

**Reply by drogon98** [reply]
Let me know when we have it. I really need it.

**Reply by ShafSpecs** [reply]
> Let me know when we have it. I really need it.

https://github.com/remix-pwa/monorepo/pull/237

Should be out in a few minutes 👍
